SDBlockDevice.cpp

```diff
@@ -558,7 +558,15 @@ int SDBlockDevice::erase(bd_addr_t addr, bd_size_t size)
     }
     int status = BD_ERROR_OK;
 
-    size -= _block_size;
+    // Toss out erases that are too small, this is fine as the block device api
+    // doesn't garuntee the final state of erased bytes and a following program
+    // will still work fine.
+    if (size < _erase_size) {
+        _lock.unlock();
+        return status;
+    }
+
+    size -= _erase_size;
```

Comment:
This should be block_size. SD card needs start and end block address to perform erase, hence size of block(last) is deducted to calculate end block address properly.